hoot-core-test/src/test/cpp/hoot/core/util/HttpTestServer.h

```diff
@@ -132,16 +132,18 @@ class HttpTestServer
    * @return True if the server should continue to listen for connections, false to stop
    */
   virtual bool respond(HttpConnection::HttpConnectionPtr& connection);
-  /** Read the HTTP headers from the request and return them, called by the overridden respond() function */
-  std::string read_request_headers(HttpConnection::HttpConnectionPtr& connection);
-  /** Read the HTTP body (after reading the request headers, called by the overridden respond() function */
-  std::string read_request_body(const std::string& headers, HttpConnection::HttpConnectionPtr& connection);
```

Comment:
Reading the headers and the body are now combined into `parse_request` because the header read sometimes was reading part of the body and throwing it away.